packages/opencensus-core/src/metrics/gauges/derived-gauge.ts

```diff
@@ -159,16 +163,18 @@ export class DerivedGauge implements types.Meter {
       throw new Error(DerivedGauge.ERROR_MESSAGE_INVALID_SIZE);
     }
 
-    if (DerivedGauge.isToValueInterface(obj)) {
-      this.extractor = () => obj.getValue();
-    } else if (DerivedGauge.isLengthAttributeInterface(obj)) {
-      this.extractor = () => obj.length;
-    } else if (DerivedGauge.isLengthMethodInterface(obj)) {
-      this.extractor = () => obj.length();
-    } else if (DerivedGauge.isSizeAttributeInterface(obj)) {
-      this.extractor = () => obj.size;
-    } else if (DerivedGauge.isSizeMethodInterface(obj)) {
-      this.extractor = () => obj.size();
+    if (typeof objOrFn === 'function') {
+      this.extractor = () => objOrFn();
```

Comment:
Can you just say `this.extractor = objOrFun`?

Reply:
To allow this I have to add new type (`ValueExtractorOrFunction`) which is type of `Function|ValueExtractor`. Let me know if this is what is desired.

Reply:
Why is that? I would have assumed that the internal `extractor` variable would always of type `() => number` (a function that returns a number). Then these different if statements are mapping from the various shapes of input to a simple function that returns a number. So in the case when the input is a function, could we just assign it directly?

Or am I missing something here?

Reply:
Defining a strongly-typed function solved this problem.

Reply:
Cool!